src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -0,0 +1,287 @@
+        featureMap = await self._read_attribute_expect_success(
+          kRootEndpointId,
+          basicinfoCluster,
+          basicinfoAttributes.FeatureMap,
+          self.dut_node_id
+        )
+
+        self.step(2)
+        # Confirms commissioning of ICD as it reads basic info from ICD device
+        reachable = await self._read_attribute_expect_success(
+          kRootEndpointId,
+          basicinfoCluster,
+          basicinfoAttributes.FeatureMap,
+          self.icd_nodeid
+        )
+
+        self.step(3)
+        # Test 1A
+        idle_mode_duration = await self._read_attribute_expect_success(
+          kICDMEndpointId,
+          icdmCluster,
+          icdmAttributes.IdleModeDuration,
+          self.icd_nodeid
+        )
+        logging.info(f"IdleModeDuration: {idle_mode_duration}")
+
+        active_mode_duration = await self._read_attribute_expect_success(
+          kICDMEndpointId,
+          icdmCluster,
+          icdmAttributes.ActiveModeDuration,
+          self.icd_nodeid
+        )
+        logging.info(f"ActiveModeDuration: {active_mode_duration}")
+
+        self.step(4)
+        # Test 1B
+
+        # Subscription to ActiveChanged
+        event = brbinfoCluster.Events.ActiveChanged
+        self.q = queue.Queue()
+        urgent = 1
+        cb = SimpleEventCallback("ActiveChanged", event.cluster_id, event.event_id, self.q)
+        subscription = await self.default_controller.ReadEvent(nodeid=self.dut_node_id, events=[(kICDBridgedEndpointId, event, urgent)], reportInterval=[1, 3])
+        subscription.SetEventUpdateCallback(callback=cb)
+
+        StayActiveDuration = 1000
+        logging.info(f"Sending KeepActiveCommand({StayActiveDuration}ms)")
+        self._send_keep_active_command(StayActiveDuration)
+
+        logging.info(f"Waiting for ActiveChanged from DUT...")
+        PromisedActiveDuration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, StayActiveDuration))/1000)
+
+        asserts.assert_true(PromisedActiveDuration >= StayActiveDuration, "PromisedActiveDuration < StayActiveDuration")
+
+        self.step(5)
+        # Test 2
+
+        StayActiveDuration = 1500
+        logging.info(f"Sending KeepActiveCommand({StayActiveDuration}ms)")
+        self._send_keep_active_command(StayActiveDuration)
+
+        logging.info(f"Waiting for ActiveChanged from DUT...")
+        PromisedActiveDuration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, StayActiveDuration))/1000)
+
+        # wait for active time duration
+        time.sleep(max(StayActiveDuration/1000, active_mode_duration))
+        # ICD now should be in idle mode
+
+        # sends 3x keep active commands
+        logging.info(f"Sending KeepActiveCommand({StayActiveDuration})")
+        self._send_keep_active_command(StayActiveDuration)
+        time.sleep(StayActiveDuration/1000)
```

Comment:
test spec says you should sleep for 100 ms here why is this a function of `StayActiveDuration`

Reply:
good catch